tests/test_other.py

```diff
@@ -9069,3 +9069,26 @@ def test_link(self):
     rval = send (0, thing, 0, 0);
     return 0;
 }''', '', force_c=True)
+
+  # This test verifies that function names embedded into the build with --js-library (JS functions imported to asm.js/wasm)
+  # are minified when -O3 is used
+  def test_js_function_imports_are_minified(self):
+    def check_size(f, expected_size):
+      if not os.path.isfile(f):
+        return # Nonexistent file passes in this check
+      obtained_size = os.path.getsize(f)
+      print('size of generated ' + f + ': ' + str(obtained_size))
+      try_delete(f)
+      assert obtained_size < expected_size
+
+    run_process([PYTHON, path_from_root('tests', 'gen_many_js_functions.py'), 'library_long.js', 'main_long.c'])
+    for wasm in [['-s', 'WASM=0', '--separate-asm']]: # TODO: Add support to Wasm to minify imports, and then add Wasm testing ['-s', 'WASM=1'] to this list
+      for closure in [['--closure', '1']]: # Currently we rely on Closure for good minification of the JS function names. TODO: Add minification also for non-Closure users and add [] to this list to test minification without Closure.
+        args = [PYTHON, EMCC, '-O3', '--js-library', 'library_long.js', 'main_long.c', '-o', 'a.html'] + wasm + closure
+        print(' '.join(args))
+        run_process(args)
+        check_size('a.js', 80000)
+        check_size('a.wasm', 80000)
+        check_size('a.asm.js', 80000)
+    try_delete('library_long.js')
+    try_delete('main_long.c')
```

Comment:
No need to clean these up, the test framework should cleanup anything files you created in CWD.

Reply:
It sounds like this kind of change has made `EMTEST_DETECT_TEMPFILE_LEAKS` check be a nop, which is supposed to verify when running tests that Emscripten does not leak test files behind. If we automatically clean all test files, we cannot test for leaks anymore.

Deleted the two lines, not sure if `EMTEST_DETECT_TEMPFILE_LEAKS` is no longer a thing.

Reply:
EMTEST_DETECT_TEMPFILE_LEAKS doesn't include changes you make in your own directory.

tearDown does this before it runs the checks:

```
       os.chdir(os.path.join(self.get_dir(), '..'))
       try_delete(self.get_dir())
```

Seems like its always worked that way.

I think this makes sense, we are more interested in stuff that is left in TMPDIR, and asking all tests to clean up after themselves adds a lot of noise the test code.